File: SoftUniCourses/SoftUniBasics/whileloops/Vacation.java
package whileloops;

import java.util.Scanner;

public class Vacation {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        //четем.нуж.пари и налични;
        double needMoney = Double.parseDouble(scanner.nextLine());
        double availabMoney = Double.parseDouble(scanner.nextLine());
        // спираме- ако харчи  5 послед. пари или стигнем до сумата,

        int days = 0;//бр.на дните за спест или  хар.
        int spendDays = 0;//бр. дни за харч.
        while (needMoney > availabMoney){
            //харчим/spend или пестим/save

            String action = scanner.nextLine();//действието четем
            double actionMoney = Double.parseDouble(scanner.nextLine());//парите от това действие
            days++;

            if (action.equals("spend")){
                spendDays++;
                availabMoney -= actionMoney;
                if (availabMoney < 0){
                    availabMoney = 0;
                }
                if (spendDays == 5){
                    System.out.printf("You can't save the money.%n" + "%d",days);
                    break;
                }

            } else if (action.equals("save")){
                spendDays = 0;
                availabMoney += actionMoney;
            }

        }

        if (needMoney <= availabMoney){
            System.out.printf("You saved the money for %d days.",days);


        }

    }
}

File: SoftUniCourses/JavaScript/VSCode/JavaScript/whileExe/vacation.js
function vacation(input) {

    let i = 0;
    let needMoney = Number(input[i]);
    i++;
    let money = Number(input[i]);
    i++;

    let counterDay = 0;// изминали дни само харчи
    let days = 0;// total брой изминали дни

    while (money < needMoney) {
        days++;
        let action = input[i];
        i++;
        let moneySpend = Number(input[i]);// харчи част от парите 
        i++;

        if (action === "spend") {
            money = money - moneySpend;
            if (money < 0) {
                money = 0;
            }
            counterDay++;
        } else {
            money = money + moneySpend;
            counterDay = 0;
        }

        if (counterDay === 5) {
            console.log("You can't save the money.");
            console.log(days);
            break
        }

    }
    if (money >= needMoney) {
        console.log(`You saved the money for ${days} days.`);

    }

}
//vacation(["110","60","spend","10","spend","10","spend","10","spend","10","spend","10"])
//vacation(["250","150","spend","50","spend","50","save","100","save","100"])

//vacation(["2000", "1000", "spend", "1200", "save", "2000"]);